x-pack/plugins/ml/server/models/data_frame_analytics/validation.ts

```diff
@@ -69,6 +78,67 @@ const analysisFieldsWarningMessage = {
   heading: analysisFieldsHeading,
 };
 
+function getRegressionAndClassificationMessage(
+  analysisConfig: AnalysisConfig,
+  analysisType: string,
+  totalDocs: number,
+  depVarCardinality: number | undefined
+) {
+  const messages = [];
+  if (isRegressionAnalysis(analysisConfig) || isClassificationAnalysis(analysisConfig)) {
+    const trainingPercent = analysisConfig[analysisType].training_percent;
+    const featureImportance = analysisConfig[analysisType].num_top_feature_importance_values;
+    // @ts-ignore num_top_classes doesn't exist on Regression type
+    const topClasses: number | undefined = analysisConfig[analysisType].num_top_classes;
+
+    if (trainingPercent) {
+      const trainingDocs = totalDocs * (trainingPercent / 100);
+      const trainingPercentMessage = getTrainingPercentMessage(trainingDocs);
+      if (trainingPercentMessage) {
+        messages.push(trainingPercentMessage);
+      }
+    }
+
+    if (featureImportance && featureImportance > 0) {
+      messages.push({
+        id: 'feature_importance',
+        text: i18n.translate(
+          'xpack.ml.models.dfaValidation.messages.featureImportanceWarningMessage',
+          {
+            defaultMessage:
+              'Enabling feature importance can result in long running jobs for jobs with a large number of training documents.',
+          }
+        ),
+        status: VALIDATION_STATUS.WARNING,
+        heading: i18n.translate('xpack.ml.models.dfaValidation.messages.featureImportanceHeading', {
+          defaultMessage: 'Feature importance',
+        }),
+      });
+    }
+
+    if (
+      (topClasses === ALL_CATEGORIES &&
+        depVarCardinality &&
+        depVarCardinality > NUM_CATEGORIES_THRESHOLD) ||
+      (topClasses && topClasses > NUM_CATEGORIES_THRESHOLD)
+    ) {
+      messages.push({
+        id: 'num_top_classes',
+        text: i18n.translate('xpack.ml.models.dfaValidation.messages.topClassesWarningMessage', {
+          defaultMessage:
+            'Probabilities will be reported for {numCategories} categories. There could be a significant effect on the size of your destination index.',
+          values: { numCategories: depVarCardinality },
```

Comment:
For the case where `topClasses > NUM_CATEGORIES_THRESHOLD` I think this should be `topClasses` and not `depVarCardinality`. Here my `depVarCardinality` is `12` but I have entered `11` for Top classes.

![image](https://user-images.githubusercontent.com/7405507/110631435-6be64100-819e-11eb-9a00-eb22944eeea2.png)